### PR TITLE
[DRAFT] Libp2p metrics

### PR DESCRIPTION
**Pull request by robschleusner**

### CODE REVIEW

File: ethdb/leveldb/leveldb.go [hunk 1]
```diff
@@ -136,11 +136,11 @@ func (db *Database) Close() error {
 	defer db.quitLock.Unlock()
 
 	if db.quitChan != nil {
-		errc := make(chan error)
-		db.quitChan <- errc
-		if err := <-errc; err != nil {
-			db.logger.Error("Metrics collection failed", "err", err)
-		}
+		// errc := make(chan error)
```
Comment (gameofpointers) [inline ethdb/leveldb/leveldb.go]: no comments, remove it, you can add it in next commit

Comment (robschleusner) [reply ethdb/leveldb/leveldb.go]: Fixed

File: quaiclient/quaiclient.go [hunk 2]
```diff
@@ -37,11 +37,11 @@ type Client struct {
 }
 
 // Dial connects a client to the given URL.
-func Dial(rawurl string) (*Client, error) {
-	return DialContext(context.Background(), rawurl)
+func Dial(rawurl string, logger log.Logger) (*Client, error) {
```
Comment (robschleusner) [inline quaiclient/quaiclient.go:40]: @gameofpointers This is causing a bug in miner where Miner needs to pass a logger in

File: quai/quaiconfig/config.go [hunk 3]
```diff
@@ -194,21 +194,21 @@ func CreateProgpowConsensusEngine(stack *node.Node, nodeLocation common.Location
...
 		MinDifficulty: config.MinDifficulty,
-	}, notify, noverify)
+	}, notify, noverify, logger)
 	engine.SetThreads(-1) // Disable CPU mining
 	return engine
 }
 
 // CreateBlake3ConsensusEngine creates a progpow consensus engine for the given chain configuration.
-func CreateBlake3ConsensusEngine(stack *node.Node, nodeLocation common.Location, config *blake3pow.Config, notify []string, noverify bool, db ethdb.Database) consensus.Engine {
+func CreateBlake3ConsensusEngine(stack *node.Node, nodeLocation common.Location, config *blake3pow.Config, notify []string, noverify bool, db ethdb.Database, logger log.Logger) consensus.Engine {
```
Comment (robschleusner) [inline quai/quaiconfig/config.go:205]: @gameofpointers This is causing a bug in miner where Miner needs to pass a logger in

**Comment by robschleusner** [comment]
Leaving this branch (libp2p-metrics) open with old logger implementation. Closing this PR and replacing it with PR #37 which implements logrus logger instead

**Pull request closed without merging**